### PR TITLE
DEP: drop python 2 support and associated ci mods

### LINKED ISSUE #1876: DEP: drop Python 2.7 support
The timeline for dropping Python 2.7 support for new Xarray releases is the end of 2018.

This issue can be used to track the necessary documentation and code changes to make that happen. 

xref: #1830

### PR DESCRIPTION
**Pull request by jhamman**
This is a WIP. I expect the CI changes to take a few iterations.

 - [x] Closes #1876
 - [x] Tests added
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

### CODE REVIEW

File: ci/requirements-py36.yml [hunk 1]
```diff
@@ -22,6 +22,9 @@ dependencies:
   - zarr
   - pseudonetcdf>=3.0.1
   - eccodes
+  # - cdat-lite  # No py3 support
+  - pynio
+  - iris>=1.10
```
Comment (jhamman) [inline ci/requirements-py36.yml]: neither pynio or iris have a py37 build yet. cdat-lite is py2 only.

Comment (shoyer) [reply ci/requirements-py36.yml]: It looks like we can get just the `cdms2` module that xarray needs from conda-forge (which does have builds for Python 3.6).

File: ci/requirements-py35-min.yml [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+  - python=3.5
   - pytest
   - flake8
   - mock
   - numpy=1.12
   - pandas=0.19
   - pip:
     - coveralls
-    - pytest-cov
+    - pytest-cov
```
Comment (shoyer) [inline ci/requirements-py35-min.yml]: nit: add the missing new-line at the end.

**Comment by shoyer** [comment]
Looks like we need to set the environment variable `UVCDAT_ANONYMOUS_LOG=no` for the cdms2 tests to work:
```
xarray/tests/test_dataarray.py::TestDataArray::test_to_and_from_cdms2_classic FAILED [ 55%]
xarray/tests/test_dataarray.py::TestDataArray::test_to_and_from_cdms2_sgrid FAILED [ 55%]
xarray/tests/test_dataarray.py::TestDataArray::test_to_and_from_cdms2_ugrid FAILED [ 55%]
```
```
Allow anonymous logging usage to help improve UV-CDAT(you can also set the environment variable UVCDAT_ANONYMOUS_LOG to yes or no)? [yes]/no: 
```

**Comment by jhamman** [comment]
@shoyer - thanks. We're all green here.

**Comment by shoyer** [comment]
Very nice, thanks!

**Pull request merged**